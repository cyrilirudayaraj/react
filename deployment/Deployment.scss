@import '~@athena/forge/sass/forge-abstracts';

.deployment {
  width: calc(100% - 75px);
  .deployment-inprogress-banner {
    padding-left: 10px;
    padding-bottom: 15px;
    width: 30%;

    .banner-header {
      display: flex;
    }
    .fe_c_banner-item__contents {
      padding-left: 50px;
      padding-top: 3px;
    }
  }
  .deployment-header {
    padding-left: 30px;
    padding-top: 5px;
    padding-bottom: 20px;
    font-size: x-large;
    font-weight: 400;
  }
}

.deployment-signoff {
  .fe_c_checkbox.fe_c_form-field__input {
    max-width: none !important;
  }
  .fe_l_grid-row__col {
    width: 100% !important;
    min-width: 100%;
    padding-right: 25px;
  }
  #releaseVersion {
    width: 30%;
  }
  .deployment-status {
    color: #008000;
    text-align: center;
  }
}

.deployment-reject {
  .fe_c_lightbox__header {
    margin-left: 10px;
  }
  .reject-status {
    color: #ca3f0d;
    text-align: center;
  }
  .fe_l_grid-row__col {
    .fe_c_form-field__input-slot {
      min-width: 100% !important;
      max-width: 100%;
      .fe_c_input,
      .fe_c_textarea {
        width: 100% !important;
      }
      .fe_c_textarea {
        height: 88px !important;
      }
    }
  }
}
