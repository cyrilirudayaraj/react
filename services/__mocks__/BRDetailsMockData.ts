import { BRDetails } from '../../types';

export const BRDetailsMockData1: BRDetails = {
  ruleReportingCategoryId: '9',
  rules2TransformationStatus: '',
  internalFixText2:
    '<h2 style="text-align:left;"><span style="color: rgb(0,0,0);background-color: rgb(255,255,255);font-size: 24px;font-family: DauphinPlain;">Where can I get some?</span></h2>\n<p style="text-align:justify;"><span style="color: rgb(0,0,0);background-color: rgb(255,255,255);font-size: 14px;font-family: Open Sans", Arial, sans-serif;">There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don\'t look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn\'t anything embarrassing hidden in the middle of text. All the Lorem Ipsum generators on the Internet tend to repeat predefined chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum which looks reasonable. The generated Lorem Ipsum is therefore always free from repetition, injected humour, or non-characteristic words etc.</span>&nbsp;</p>',
  ruleTypeId: '3',
  internalFixText1:
    '<p><strong>Lorem Ipsum</strong> <span style="color: rgb(0,0,0);background-color: rgb(255,255,255);font-size: 14px;font-family: Open Sans;">is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.</span>&nbsp;</p>',
  visitRuleDisplayLocation: 'VISITBILL_MEDICALCODES',
  contextName: 'CA - Prime Care - Inland Valley',
  contextId: '21',
  id: '1251',
  visitRuleDisplayLocationId: '2',
  businessRequirementTypeId: '2',
  legacyRuleId: '',
  rules2TransformationStatusId: '',
  name: 'Business Requirement Name',
  ruleReportingCategoryName: 'Coding: Procedure Code',
  description:
    '<p><span style="color: rgb(95,95,95);background-color: rgb(255,255,255);font-size: 14px;font-family: Source Sans Pro", arial, sans-serif;">Business Requirement Description (Client Friendly)</span>&nbsp;</p>',
  ruleType: 'CLAIMCREATE',
  localRuleUseCaseId: '2',
  localRuleUseCaseName: 'Capitation',
  businessRequirementType: 'Local',
  associatedBRTasks: [
    {
      id: '12',
      taskName: 'test1',
      statusId: '2',
      priorityId: '1',
      statusName: 'In Progress',
      escalatedYn: 'N',
      deploymentDate: '07/28/2021',
    },
    {
      id: '13',
      taskName: 'test2',
      statusId: '3',
      priorityId: '3',
      statusName: 'Assigned',
      escalatedYn: 'Y',
      deploymentDate: '07/29/2021',
    },
  ],
};
