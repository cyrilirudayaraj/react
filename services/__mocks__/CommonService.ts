import { cloneDeep } from 'lodash';
import { AuditEvent, TaskDetail } from '../../types';
import { MockData } from './MockData';
import { TaskDetailsMockData1 } from './TaskDetailsMockData';
import { TaskHistoryMockData } from './TaskHistoryMockData';
import { DeploymentData1 } from './DeploymentMockData';
import AppConstants from '../../constants/AppConstants';
import { BRDetailsMockData1 } from './BRDetailsMockData';

export async function getPriorities() {
  return await new Promise((resolve) => {
    resolve(MockData.PRIORITIES);
  });
}

export async function getPrioritiesWithSla() {
  return await new Promise((resolve) => {
    resolve(MockData.PRIORITIES);
  });
}

export async function getOriginatingSystems() {
  return await new Promise((resolve) => {
    resolve(MockData.ORGINATINGSYSTEMS);
  });
}

export async function getTaskTypes() {
  return await new Promise((resolve) => {
    resolve(MockData.TASKTYPES);
  });
}

export async function getUsers() {
  return await new Promise((resolve) => {
    resolve(MockData.USERS);
  });
}

export async function searchBusinessRequirements(legacyRulePrefix = '') {
  return await new Promise((resolve) => {
    resolve(MockData.BUSINESSREQUIREMENTS);
  });
}

export async function brLookup(businessRequirementId = '') {
  const task = {
    taskId: 19,
  };

  return await new Promise((resolve) => {
    resolve(task);
  });
}

export async function createTask(payload: any) {
  return await new Promise((resolve, reject) => {
    if (payload['PRIORITYID'] == 1) {
      resolve({ id: '000001' });
    } else {
      reject({ message: 'Request failed' });
    }
  });
}

export async function getTaskList(payload: any) {
  const taskList = [
    {
      activeTaskStepId: '117',
      assignedTo: 'aanandkumar',
      businessRequirementId: '5',
      dueOn: '07/02/2020',
      lastModified: '07/02/2020',
      legacyRuleId: '121212',
      priorityName: 'P1',
      statusName: 'Assigned',
      taskId: '28',
      taskName:
        'Update Global Business Requirement 123456:12 Exclude XHC Claim Task1',
      taskTypeId: '3',
      taskTypeName: 'Dual Update',
      workflowStepName: 'Analyze Task',
    },
  ];

  if (payload && payload.searchText && payload.searchText === 'Update') {
    return await new Promise((resolve) => {
      resolve({
        limit: 10,
        offset: 0,
        result: taskList,
        total: 25,
      });
    });
  }
  if (payload && payload.statusIds && payload.statusIds[0] === '10') {
    return await new Promise((resolve) => {
      resolve({
        limit: 10,
        offset: 0,
        result: taskList,
        total: 25,
      });
    });
  }
  return await new Promise((resolve) => {
    resolve({
      limit: 10,
      offset: 0,
      result: [],
      total: 0,
    });
  });
}

export function getTaskCountByQueue() {
  return new Promise((resolve) => {
    resolve(MockData.QUEUETASKCOUNT);
  });
}

export async function getTaskDetails(id: string) {
  return await new Promise((resolve) => {
    resolve(TaskDetailsMockData1);
  });
}

export function updateTask(payload: any): Promise<TaskDetail> {
  return new Promise((resolve) => {
    resolve(TaskDetailsMockData1);
  });
}

export function getBusinessRequirementTypes() {
  return new Promise((resolve) => {
    resolve(MockData.BUSINESSREQUIREMENTTYPES);
  });
}

export function getRuleReportingCategories() {
  return new Promise((resolve) => {
    resolve(MockData.RULEREPORTINGCATEGORIES);
  });
}

export function getUserPermissions() {
  return new Promise((resolve) => {
    resolve(MockData.USERPERMISSIONS);
  });
}

export function getContextList(context = '') {
  return new Promise((resolve) => {
    resolve(MockData.CONTEXTS);
  });
}

export function getLocalRuleUseCaseList() {
  return new Promise((resolve) => {
    resolve(MockData.LOCALRULEUSECASES);
  });
}

export function getDepartmentOrigins() {
  return new Promise((resolve) => {
    resolve(MockData.DEPARTMENTORIGINS);
  });
}

export async function getRuleEngines() {
  return new Promise((resolve) => {
    resolve(MockData.RULEENGINES);
  });
}

export function getRuleTypes() {
  return new Promise((resolve) => {
    resolve(MockData.RULETYPES);
  });
}

export function getVisitRuleDisplayLocations() {
  return new Promise((resolve) => {
    resolve(MockData.VISITRULEDISPLAYLOCATIONS);
  });
}

export async function getDependencySystems() {
  return new Promise((resolve) => {
    resolve(MockData.DEPENDENCYSYSTEMS);
  });
}

export async function getAttachmentTypes() {
  return await new Promise((resolve) => {
    resolve(MockData.ATTACHMENTTYPES);
  });
}

export async function getReportTypes() {
  return await new Promise((resolve) => {
    resolve(MockData.REPORTTYPES);
  });
}

export async function downloadReport(payload: any) {
  return await new Promise((resolve, reject) => {
    if (payload['REPORTTYPEID'] == 1) {
      resolve({ id: '000001' });
    } else {
      reject({ message: 'Request failed' });
    }
  });
}

export function getWorkFlowSteps() {
  return new Promise((resolve) => {
    resolve(MockData.WORKFLOWSTEPS);
  });
}

export function getStatus() {
  return new Promise((resolve) => {
    resolve(MockData.STATUSES);
  });
}

export function getAttachmentDetails(taskId: string) {
  return new Promise((resolve) => {
    resolve(MockData.ATTACHMENTS);
  });
}

export function createAttachment(payload: any) {
  return new Promise((resolve) => {
    resolve(MockData.ATTACHMENTS);
  });
}

export function updateAttachmentDetails(payload: any) {
  return new Promise((resolve) => {
    resolve(MockData.ATTACHMENTS);
  });
}

export function deleteAttachmentDetails(payload: any) {
  return new Promise((resolve) => {
    resolve(MockData.ATTACHMENTS);
  });
}

export function getTimeEntryLogs(taskId: string) {
  return new Promise((resolve) => {
    resolve(MockData.TIMEENTRYLOGS);
  });
}

export function createTimeEntry(payload: any) {
  return new Promise((resolve) => {
    resolve(MockData.TIMEENTRYLOGS);
  });
}
export function updateTimeEntry(payload: any) {
  return new Promise((resolve) => {
    resolve(MockData.TIMEENTRYLOGS);
  });
}
export function deleteTimeEntry(payload: any) {
  return new Promise((resolve) => {
    resolve([]);
  });
}

export function getUserComments(taskId: string) {
  return new Promise((resolve) => {
    resolve(MockData.USERCOMMENTS);
  });
}

export function createUserCommentDetail(payload: any) {
  return new Promise((resolve) => {
    resolve(MockData.USERCOMMENTS);
  });
}

export function getTaskHistory(taskId: string): Promise<AuditEvent[]> {
  return new Promise((resolve) => {
    resolve(TaskHistoryMockData);
  });
}

export function fetchDependencyDetailsListByIds(payload: any) {
  return new Promise((resolve) => {
    const [firstDependency] = payload;

    let mockDependencies = MockData.DEPENDENCIES;
    if (firstDependency && firstDependency.dependencyId == '1') {
      mockDependencies = cloneDeep(mockDependencies);
      mockDependencies[0].activeWorkflowStepId = '7';
    }
    if (payload) resolve(mockDependencies);
  });
}

export function getSalesforceCaseDetails(caseId: string): Promise<any> {
  return new Promise((resolve) => {
    resolve({
      subject: 'claim',
      practiceId: '951',
      id: '5006f00001cJJsGAAW',
      salesforceId: '11309869',
    });
  });
}

export function getDeployments(payload: any): Promise<any> {
  if (payload.LIMIT == 1) {
    return new Promise((resolve) => {
      resolve([
        {
          id: '1',
          status: 'SUCCESS',
          releaseVersion: '2.0.1',
          createdBy: 'jcyril',
        },
      ]);
    });
  } else {
    return new Promise((resolve) => {
      resolve(DeploymentData1);
    });
  }
}

export function deploy(payload: any) {
  return new Promise((resolve) => {
    resolve(DeploymentData1);
  });
}

export function completeDeployment(payload: any) {
  return new Promise((resolve) => {
    const response = cloneDeep(DeploymentData1);
    response.statusId =
      AppConstants.SERVER_CONSTANTS.DEPLOYMENT_STATUSES.COMPLETE;
    response.status = 'COMPLETE';
    resolve(response);
  });
}

export function rejectDeployment(payload: any): Promise<any> {
  return new Promise((resolve) => {
    const response = cloneDeep(DeploymentData1);
    response.statusId =
      AppConstants.SERVER_CONSTANTS.DEPLOYMENT_STATUSES.REJECTED;
    response.status = 'REJECTED';
    resolve(response);
  });
}
export function getDeploymentStatuses(payload: any): Promise<any> {
  return new Promise((resolve) => {
    resolve(MockData.DEPLOYMENTSTATUSES);
  });
}

export async function getScrubTypes() {
  return await new Promise((resolve) => {
    resolve(MockData.SCRUBTYPELIST);
  });
}

export async function getPriorityReasons() {
  return await new Promise((resolve) => {
    resolve(MockData.PRIORITYREASONLIST);
  });
}

export async function getClaimRuleCategoryList() {
  return await new Promise((resolve) => {
    resolve(MockData.CLAIMRULECATEGORYLIST);
  });
}
export async function searchMigrationRules(payload: any) {
  if (payload && payload.searchText && payload.searchText === '58') {
    return await new Promise((resolve) => {
      resolve(MockData.MIGRATIONRULESLIST);
    });
  } else {
    return await new Promise((resolve) => {
      resolve({
        limit: 10,
        offset: 0,
        result: [],
        total: 0,
      });
    });
  }
}
export async function getRationalizationRules(payload: any) {
  return await new Promise((resolve) => {
    resolve(MockData.RATIONALIZATIONRULES);
  });
}

export async function getBusinessRequirementDetails(id: string) {
  return await new Promise((resolve) => {
    resolve(BRDetailsMockData1);
  });
}
