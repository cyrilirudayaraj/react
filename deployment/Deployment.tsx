import { Banner, BannerItem, TabPane, Tabs } from '@athena/forge';
import React, { useEffect, useState } from 'react';
import Labels from '../../constants/Labels';
import DeploymentList from './deployment-list/DeploymentList';
import {
  getDeploymentStatuses,
  fetchDeploymentStatusesOnce,
} from '../../slices/DeploymentSlice';
import { connect } from 'react-redux';
import { fetchUsersOnce, getUsers } from '../../slices/MasterDataSlice';
import { cloneDeep, isEmpty } from 'lodash';
import ConversionUtil from '../../utils/ConversionUtil';

import DeploymentTaskList from './deployment-task-list/DeploymentTaskList';
import AppConstants from '../../constants/AppConstants';
import { getDeployments } from '../../services/CommonService';
import './Deployment.scss';
import StringUtil from '../../utils/StringUtil';
import { Link } from 'react-router-dom';
import { Deployment as DeploymentType } from '../../types';
import DeploymentDetailsComponent from './deployment-details/DeploymentDetails';

interface DeploymentDetailsType {
  decisionServiceName: string;
  username: string;
  password: string;
  deployment: any;
}

function Deployment(props: any): JSX.Element {
  const [selectedIndex, setSelectedIndex] = useState(0);
  const [tabIndexVal, setTabIndexVal] = useState(0);
  const [releasedByFilter, setReleasedByFilter] = useState<any>([]);
  const [statusFilter, setStatusFilter] = useState<any>([]);
  const delay = 5 * 1000 * 60;
  useEffect(() => {
    setSelectedIndex(tabIndexVal);
    props.fetchUsersOnce();
    props.fetchDeploymentStatusesOnce();
    getDeploymentInprogress();
    const interval = setInterval(() => {
      getDeploymentInprogress(AppConstants.UI_CONSTANTS.DEPLOY_CONFIG);
    }, delay);
    return () => clearInterval(interval);
  }, []);
  const getAllOptionObj = (): any => {
    return {
      userName: Labels.DEPLOYMENT_GRID.ALL,
    };
  };
  useEffect(() => {
    setSelectedIndex(tabIndexVal);
    if (!isEmpty(props.userList) && isEmpty(releasedByFilter)) {
      const userList = cloneDeep(props.userList);
      userList.unshift(getAllOptionObj());
      const userName = ConversionUtil.convertMapToDropDownList(
        userList,
        'userName'
      );
      setReleasedByFilter(userName);
    }

    if (!isEmpty(props.deploymentStatuses) && isEmpty(statusFilter)) {
      let deploymentStatusList: any[] = [];
      deploymentStatusList = props.deploymentStatuses?.map(
        (status: any) => status.name
      );
      const statusTypes = cloneDeep(deploymentStatusList);
      statusTypes.unshift('All');
      setStatusFilter(statusTypes);
    }
  });

  const [deploymentDetails, setDeploymentDetails] = useState<
    DeploymentDetailsType | undefined
  >({
    decisionServiceName: '',
    username: '',
    password: '',
    deployment: undefined,
  });

  const [isDeployed, setIsDeployed] = useState(false);
  const [deploymentInprogress, setdeploymentInprogress] = useState<
    DeploymentType[]
  >([]);

  const handleTabsChange = (e: any) => {
    const value = e.target.value;
    setTabIndex(parseInt(value));
    getDeploymentInprogress();
  };

  const setTabIndex = (index: number) => {
    setSelectedIndex(index);
    setTabIndexVal(index);
  };

  const getDeploymentInprogress = (config?: any) => {
    getDeployments(
      {
        statusIds: [
          AppConstants.SERVER_CONSTANTS.DEPLOYMENT_STATUSES.IN_PROGRESS,
        ],
      },
      config
    ).then((response: any) => {
      setdeploymentInprogress(response);
    });
  };

  const getDeploymentInprogressIds = () => {
    let formatDeploymentId = '';
    const ids: string[] = [];
    if (deploymentInprogress && deploymentInprogress.length > 0) {
      deploymentInprogress.forEach((deployment: any, index: number) => {
        ids[index] = StringUtil.formatDeploymentID(deployment.id);
      });
      formatDeploymentId = ids.join(', ');
    }
    return formatDeploymentId;
  };

  const handelDeploymentSuccessEvent = (
    deploymentDetails: DeploymentDetailsType
  ) => {
    setDeploymentDetails(deploymentDetails);
    setIsDeployed(true);
  };

  const handelBackButton = () => {
    getDeploymentInprogress();
    setDeploymentDetails(undefined);
    setIsDeployed(false);
  };

  const getBannerTemplate = () => {
    return deploymentInprogress.length < 1 ? (
      <React.Fragment></React.Fragment>
    ) : (
      <div className="deployment-inprogress-banner">
        <Banner>
          <BannerItem
            headerText={
              Labels.DEPLOYMENT_GRID.DEPLOYMENT +
              AppConstants.UI_CONSTANTS.WHITE_SPACE +
              getDeploymentInprogressIds() +
              AppConstants.UI_CONSTANTS.WHITE_SPACE +
              Labels.DEPLOY.IN_PROGRESS
            }
            {...{ className: 'banner-header' }}
          >
            <Link
              className="deployment-id-link"
              to={
                process.env.REACT_APP_BASE_CONTEXT_PATH +
                `deployments/${deploymentInprogress[0].id}`
              }
            >
              {Labels.DEPLOY.VIEW}
            </Link>
          </BannerItem>
        </Banner>
      </div>
    );
  };

  const getComponentTemplate = () => {
    return isDeployed ? (
      <DeploymentDetailsComponent
        decisionServiceName={deploymentDetails?.decisionServiceName}
        username={deploymentDetails?.username}
        password={deploymentDetails?.password}
        deployment={deploymentDetails?.deployment}
        onBack={handelBackButton}
      />
    ) : (
      <div className="fe_u_padding--top-large deployment">
        <div className="deployment-header">
          <span>{Labels.DEPLOY.DEPLOYMENT_WORKLIST}</span>
        </div>
        {getBannerTemplate()}
        <Tabs selectedIndex={selectedIndex} onTabsChange={handleTabsChange}>
          <TabPane label={Labels.DEPLOY.TASKS_AVAILABLE_TO_DEPLOY} padded>
            <DeploymentTaskList
              isDeploymentInprogress={deploymentInprogress.length > 0}
              handleDeploymentInprogressValidation={getDeploymentInprogress}
              handelDeploymentSuccessEvent={handelDeploymentSuccessEvent}
            />
          </TabPane>
          <TabPane label={Labels.DEPLOY.DEPLOYMENTS} padded>
            <DeploymentList
              statusFilter={statusFilter}
              releasedByFilter={releasedByFilter}
              deploymentStatuses={props.deploymentStatuses}
              handleDeploymentInprogressValidation={getDeploymentInprogress}
            />
          </TabPane>
        </Tabs>
      </div>
    );
  };
  return <React.Fragment>{getComponentTemplate()}</React.Fragment>;
}
const mapDispatchToProps = {
  fetchUsersOnce,
  fetchDeploymentStatusesOnce,
};
const mapStateToProps = (state: any) => {
  return {
    deploymentStatuses: getDeploymentStatuses(state),
    userList: getUsers(state),
  };
};
export default connect(mapStateToProps, mapDispatchToProps)(Deployment);
