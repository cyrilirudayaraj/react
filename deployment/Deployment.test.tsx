import React from 'react';
import { configure, mount } from 'enzyme';
import { Provider } from 'react-redux';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import Deployment from './Deployment';
import { DEPLOYMOCKDATA } from '../../services/__mocks__/MockData';

jest.mock('../../services/CommonService');
configure({ adapter: new Adapter() });
let store: any;
beforeEach(() => {
  const configs = [
    {
      textValue: 'atlas',
      dateValue: null,
      name: 'app.name',
      id: '1',
      numericValue: null,
    },

    {
      textValue: 'rules-master',
      dateValue: null,
      name: 'app.ui.package.name',
      id: '2',
      numericValue: null,
    },
  ];
  store = configureStore([])({
    masterData: {
      configs: configs,
      taskDetails: [],
      users: [],
    },
    deployment: {
      deploymentDetails: DEPLOYMOCKDATA,
      deploymentStatuses: [],
    },

    userPermission: {
      userPermissions: ['task.deploy'],
    },
    getDeploymentDetails: jest.fn(),
    rejectDeployment: jest.fn(),
  });
  store.dispatch = jest.fn();
});
describe('test <Layout>', () => {
  let wrapper: any;
  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <Deployment />
      </Provider>
    );
  });
  it('should render Header element', () => {
    expect(wrapper.debug()).toMatchSnapshot();
    expect(wrapper.find(Deployment)).toHaveLength(1);
  });
});
