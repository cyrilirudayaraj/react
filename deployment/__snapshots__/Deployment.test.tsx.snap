// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <Layout> should render Header element 1`] = `
"<Provider store={{...}}>
  <Connect(Deployment)>
    <Deployment deploymentStatuses={{...}} userList={{...}} fetchUsersOnce={[Function (anonymous)]} fetchDeploymentStatusesOnce={[Function (anonymous)]}>
      <div className=\\"fe_u_padding--top-large deployment\\">
        <div className=\\"deployment-header\\">
          <span>
            Deployment Worklist
          </span>
        </div>
        <Tabs selectedIndex={0} onTabsChange={[Function: handleTabsChange]} defaultSelectedIndex={0}>
          <TabsControlled className={[undefined]} tabGroupLabel={[undefined]} selectedIndex={0} onTabsChange={[Function: handleTabsChange]}>
            <div className=\\"fe_c_tabs\\">
              <ul className=\\"fe_c_tabs__labels\\" role=\\"tablist\\" aria-label={[undefined]}>
                <li className=\\"fe_c_tabs__label-wrapper\\" role=\\"presentation\\" style={{...}}>
                  <button type=\\"button\\" role=\\"tab\\" value={0} aria-selected={true} className=\\"fe_c_tabs__label fe_is-selected\\" aria-hidden={false} onClick={[Function: bound handleTabLabelClick]} onKeyDown={[Function: bound handleTabLabelKeypress]} data-content=\\"Tasks available to deploy\\" tabIndex=\\"0\\">
                    <div className=\\"fe_c_tabs__label-text\\">
                      Tasks available to deploy
                    </div>
                  </button>
                </li>
                <li className=\\"fe_c_tabs__label-wrapper\\" role=\\"presentation\\" style={{...}}>
                  <button type=\\"button\\" role=\\"tab\\" value={1} aria-selected={false} className=\\"fe_c_tabs__label\\" aria-hidden={false} onClick={[Function: bound handleTabLabelClick]} onKeyDown={[Function: bound handleTabLabelKeypress]} data-content=\\"Deployments\\" tabIndex=\\"-1\\">
                    <div className=\\"fe_c_tabs__label-text\\">
                      Deployments
                    </div>
                  </button>
                </li>
              </ul>
              <div className=\\"fe_c_tabs__shadow\\" />
              <div className=\\"fe_c_tabs__pane-wrapper\\">
                <TabPane label=\\"Tasks available to deploy\\" padded={true} mountedWhileHidden={false} show={true}>
                  <div label=\\"Tasks available to deploy\\" style={{...}} className=\\"fe_c_tab-pane fe_c_tab-pane--padded\\" role=\\"tabpanel\\">
                    <Connect(DeploymentTaskList) isDeploymentInprogress={false} handleDeploymentInprogressValidation={[Function: getDeploymentInprogress]} handelDeploymentSuccessEvent={[Function: handelDeploymentSuccessEvent]}>
                      <DeploymentTaskList isDeploymentInprogress={false} handleDeploymentInprogressValidation={[Function: getDeploymentInprogress]} handelDeploymentSuccessEvent={[Function: handelDeploymentSuccessEvent]} decisionServiceInfo={[undefined]} dispatch={[Function: mockConstructor]}>
                        <div className=\\"deployment-task-list\\">
                          <div>
                            <Connect(WFShowForPermission) permission=\\"task.deploy\\">
                              <WFShowForPermission permission=\\"task.deploy\\" userPermissions={{...}} dispatch={[Function: mockConstructor]}>
                                <NewDeployment disabled={true} onDeploy={[Function (anonymous)]} selectedTasks={{...}} decisionServiceName={[undefined]}>
                                  <ForwardRef text=\\"Deploy\\" variant=\\"primary\\" disabled={true} onClick={[Function (anonymous)]} fullWidth={false} iconSet={{...}} size=\\"medium\\" showLoadingSpinner={false} type=\\"button\\" useLink={false}>
                                    <button className=\\"fe_c_button fe_c_button--primary fe_c_button--medium fe_is-disabled\\" aria-disabled={true} aria-label={[undefined]} onClick={[Function: handleClick]} type=\\"button\\">
                                      <div className=\\"fe_c_button__content\\">
                                        <span className=\\"fe_c_button__text fe_c_button__text--medium fe_c_button__text--primary\\">
                                          Deploy
                                        </span>
                                      </div>
                                    </button>
                                  </ForwardRef>
                                </NewDeployment>
                              </WFShowForPermission>
                            </Connect(WFShowForPermission)>
                          </div>
                          <div className=\\"table-style\\">
                            <DeploymentTaskListGrid dataSource={{...}} onSelectAllCheckbox={[Function (anonymous)]} onSelectCheckbox={[Function (anonymous)]}>
                              <Table layout=\\"compact\\" className=\\"full-width\\" rows={{...}} rowKey=\\"taskId\\" columns={{...}} showHover={true} onRow={[Function: onRow]} replaceChildren={false} nowrapHeaders={false}>
                                <Provider className=\\"fe_c_table fe_c_table--show-hover fe_c_table--compact full-width\\" columns={{...}} renderers={{...}}>
                                  <table className=\\"fe_c_table fe_c_table--show-hover fe_c_table--compact full-width\\">
                                    <Header>
                                      <thead>
                                        <HeaderRow renderers={{...}} onRow={[Function: onRow]} rowData={{...}} rowIndex={0}>
                                          <tr>
                                            <th data-column=\\"taskId\\" scope=\\"col\\" role=\\"columnheader\\" title={{...}} className=\\"fe_c_table__header\\">
                                              <div className=\\"select-all-checkbox\\">
                                                <FormField id=\\"select-all-checkbox\\" className=\\"font-bold\\" inputAs={{...}} labelText=\\"\\" description=\\"Task ID\\" onClick={[Function: onClick]} required={false}>
                                                  <FormFieldLayout labelWidth={[undefined]} nested={[undefined]} layout={[undefined]} statusExtras={[undefined]} className=\\"font-bold\\" labelAlwaysAbove={[undefined]} labelSlot={{...}} messageSlot={[undefined]} statusIndicatorSlot={[undefined]} inputSlot={{...}}>
                                                    <div className=\\"fe_c_form-field fe_c_form-field--medium font-bold\\">
                                                      <GridRow nested={true}>
                                                        <div className=\\"fe_l_grid-row fe_l_grid-row--nested\\">
                                                          <GridCol width={{...}} className=\\"fe_c_form-field__label\\">
                                                            <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-12 fe_l_grid-row__col--medium-3 fe_c_form-field__label\\">
                                                              <Label id=\\"select-all-checkbox-label\\" htmlFor={[undefined]} text=\\"\\" disabled={[undefined]} required={false} useLegend={false}>
                                                                <label htmlFor={[undefined]} id=\\"select-all-checkbox-label\\" className=\\"fe_c_label\\" />
                                                              </Label>
                                                            </div>
                                                          </GridCol>
                                                          <GridCol width={{...}} className=\\"fe_c_form-field__right\\">
                                                            <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-12 fe_l_grid-row__col--medium-9 fe_c_form-field__right\\">
                                                              <div className=\\"fe_c_form-field__input-slot\\">
                                                                <Checkbox description=\\"Task ID\\" onClick={[Function: onClick]} className=\\"fe_c_form-field__input\\" id=\\"select-all-checkbox\\" name=\\"select-all-checkbox\\" required={false} hideRequiredStyles={false} disabled={false} error={[undefined]} aria-labelledby=\\"select-all-checkbox-label\\">
                                                                  <span className=\\"fe_c_checkbox fe_c_form-field__input\\">
                                                                    <input type=\\"checkbox\\" onClick={[Function: onClick]} name=\\"select-all-checkbox\\" className=\\"fe_c_checkbox__input\\" disabled={false} required={false} id=\\"select-all-checkbox\\" defaultChecked={[undefined]} aria-labelledby=\\"select-all-checkbox-label description-label-select-all-checkbox\\" />
                                                                    <Label htmlFor=\\"select-all-checkbox\\" className=\\"fe_c_checkbox__description\\" text=\\"Task ID\\" id=\\"description-label-select-all-checkbox\\" aria-label={[undefined]} required={false} useLegend={false}>
                                                                      <label htmlFor=\\"select-all-checkbox\\" id=\\"description-label-select-all-checkbox\\" aria-label={[undefined]} className=\\"fe_c_label fe_c_checkbox__description\\">
                                                                        Task ID
                                                                      </label>
                                                                    </Label>
                                                                  </span>
                                                                </Checkbox>
                                                              </div>
                                                              <div className=\\"fe_c_form-field__status\\" />
                                                            </div>
                                                          </GridCol>
                                                        </div>
                                                      </GridRow>
                                                    </div>
                                                  </FormFieldLayout>
                                                </FormField>
                                              </div>
                                            </th>
                                            <th data-column=\\"taskName\\" scope=\\"col\\" role=\\"columnheader\\" title={{...}} className=\\"fe_c_table__header\\">
                                              Task Name
                                            </th>
                                            <th data-column=\\"taskTypeName\\" scope=\\"col\\" role=\\"columnheader\\" title={{...}} className=\\"fe_c_table__header\\">
                                              Task Type
                                            </th>
                                            <th data-column=\\"statusName\\" scope=\\"col\\" role=\\"columnheader\\" title={{...}} className=\\"fe_c_table__header\\">
                                              Status
                                            </th>
                                            <th data-column=\\"workflowStepName\\" scope=\\"col\\" role=\\"columnheader\\" title={{...}} className=\\"fe_c_table__header\\">
                                              Workflow Step
                                            </th>
                                            <th data-column=\\"deploymentDate\\" scope=\\"col\\" role=\\"columnheader\\" title={{...}} className=\\"fe_c_table__header\\">
                                              Deployment Date
                                            </th>
                                            <th data-column=\\"completedTaskDependencies\\" scope=\\"col\\" role=\\"columnheader\\" title={{...}} className=\\"fe_c_table__header\\">
                                              Dependencies/ Related Tasks
                                            </th>
                                            <th data-column=\\"businessRequirementId\\" scope=\\"col\\" role=\\"columnheader\\" title={{...}} className=\\"fe_c_table__header\\">
                                              BR ID
                                            </th>
                                            <th data-column=\\"legacyRuleId\\" scope=\\"col\\" role=\\"columnheader\\" title={{...}} className=\\"fe_c_table__header\\">
                                              1.0 Rule ID
                                            </th>
                                          </tr>
                                        </HeaderRow>
                                      </thead>
                                    </Header>
                                    <Body onRow={[Function: onRow]} rows={{...}} rowKey=\\"taskId\\">
                                      <tbody />
                                    </Body>
                                  </table>
                                </Provider>
                              </Table>
                            </DeploymentTaskListGrid>
                          </div>
                        </div>
                      </DeploymentTaskList>
                    </Connect(DeploymentTaskList)>
                  </div>
                </TabPane>
                <TabPane label=\\"Deployments\\" padded={true} mountedWhileHidden={false} show={false} />
              </div>
            </div>
          </TabsControlled>
        </Tabs>
      </div>
    </Deployment>
  </Connect(Deployment)>
</Provider>"
`;
